src/variables.jl

```diff
@@ -55,6 +88,59 @@ function add_constrained_variable(model::ModelLike, set::AbstractScalarSet)
     return variable, constraint
 end
 
+"""
+    supports_add_constrained_variables(
+        model::ModelLike,
+        S::Type{<:AbstractScalarSet}
+    )::Bool
+
+Return a `Bool` indicating whether `model` supports constraining a vector of
+variables to belong to a set of type `S` either on creation of the vector of
+variables with [`add_constrained_variables`](@ref) or after the variable is
+created with [`add_constraint`](@ref).
+
+By default, if `S` is `Reals` then this function returns `true` and otherwise,
+it falls back to `supports_add_constrained_variables(model, Reals) &&
+supports_constraint(model, MOI.VectorOfVariables, S)` which is the correct
+definition for most models.
+
+## Example
+
+In the standard conic form, the variables are grouped into several cones
+and the constraints are affine equality constraints.
+If `Reals` is not one of the cones supported by the solvers then it needs
+to implement `supports_add_constrained_variables(::Optimizer, ::Type{Reals}) = false`
+as free variables are not supported.
+The solvers should then implement
+`supports_add_constrained_variables(::Optimizer, ::Type{<:SupportedCones}) = true`
+where `SupportedCones` is the union of all cone types that are supported
+but it should not implement
```

Comment:
This is a run-on sentence and a bit confusing. You need a comma before "but". A semi-colon would work also.
"should not implement" doesn't follow from "should return `false`". The solver could implement it to return false (but that's not necessary because it's the default). Please tidy up the logic.